modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDataTypesTest.java

```diff
@@ -655,6 +656,65 @@ public void charLimitationWithMergeUpdateOp() {
         }
     }
 
+    @Test
+    public void zeroStringsAreNotAllowed() {
+        // Char
+
+        assertThrowsSqlException(
+                STMT_VALIDATION_ERR,
+                "Length for type CHAR must be at least 1",
+                () -> sql("SELECT CAST(1 AS CHAR(0))")
+        );
+
+        assertThrowsSqlException(
+                STMT_VALIDATION_ERR,
+                "Length for type CHAR must be at least 1",
+                () -> sql("SELECT '1'::CHAR(0)")
+        );
+
+        // Varchar
+
+        assertThrowsSqlException(
+                STMT_VALIDATION_ERR,
+                "Length for type VARCHAR must be at least 1",
+                () -> sql("SELECT CAST(1 AS VARCHAR(0))")
+        );
+
+        assertThrowsSqlException(
+                STMT_VALIDATION_ERR,
+                "Length for type VARCHAR must be at least 1",
+                () -> sql("SELECT '1'::VARCHAR(0)")
+        );
+
+        // Binary
+
+        assertThrowsSqlException(
+                STMT_VALIDATION_ERR,
+                "Length for type BINARY must be at least 1",
```

Comment:
it looks strange that we have different messages for different types. Looks like we should have one template.

Reply:
I mean '.....must be at least 1' and '.....must be positive'